File: leetcode/leetcode_344_reverse_string/solution.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: List[str]
        :rtype: None Do not return anything, modify s in-place instead.
        """
        reverse(s, 0, len(s) - 1)
        

    def reverse(self, s, left, right):
        if left >= right:
            return
        s[left], s[right] = s[right], s[left]
        self.reverse(s, left + 1, right - 1)
    
    

File: leetcode/leetcode_344_reverse_string/Solution.java
package leetcode_344_reverse_string;

public class Solution {
    public void reverseString(char[] s) {
        int l = 0, r = s.length - 1;
        while (l < s.length/2) {
            char temp;
            temp = s[l];
            s[l] = s[r];
            s[r] = temp;
            l++;
            r--;

        }
    }
}
